experimental/BotBuilder-lsp-lg/server/src/server.ts

```diff
@@ -0,0 +1,242 @@
+
+// Create a simple text document manager. 
+const documents: TextDocuments<TextDocument> = new TextDocuments(TextDocument);
+let workspaceFolders: WorkspaceFolder[] | null | undefined;
+
+let hasConfigurationCapability = false;
+let hasWorkspaceFolderCapability = false;
+let hasDiagnosticRelatedInformationCapability = false;
+let hasDidChangeWatchedFilesCapability = false;
+
+
+connection.onInitialize((params: InitializeParams) => {
+	const capabilities = params.capabilities;
+	workspaceFolders = params.workspaceFolders;
+	util.triggerLGFileFinder(workspaceFolders!);
+
+	// Does the client support the `workspace/configuration` request?
+	// If not, we fall back using global settings.
+	hasConfigurationCapability = !!(
+		capabilities.workspace && !!capabilities.workspace.configuration
+	);
+	hasWorkspaceFolderCapability = !!(
+		capabilities.workspace && !!capabilities.workspace.workspaceFolders
+	);
+	hasDiagnosticRelatedInformationCapability = !!(
+		capabilities.textDocument &&
+		capabilities.textDocument.publishDiagnostics &&
+		capabilities.textDocument.publishDiagnostics.relatedInformation
+	);
+	hasDidChangeWatchedFilesCapability = !!(
+		capabilities.workspace && !! capabilities.workspace.didChangeWatchedFiles?.dynamicRegistration
+	);
+
+	const result: InitializeResult = {
+		capabilities: {
+			textDocumentSync: TextDocumentSyncKind.Incremental,
+			// Tell the client that this server supports code completion.
+			completionProvider: {
+				// resolveProvider: true,
+				triggerCharacters: [ '{', '(', '[', '.', '\n', '#', '=', ',' ]
+			},
+			hoverProvider: true,
+			definitionProvider: true,
+			executeCommandProvider: {
+				commands: ['lg.extension.onEnterKey']
+			},
+			signatureHelpProvider: {
+				triggerCharacters: ['(', ',']
+			},
+			workspace: {
+				workspaceFolders: {
+					supported: true
+				}
+			}
+		}
+	};
+	if (hasWorkspaceFolderCapability) {
+		result.capabilities.workspace = {
+			workspaceFolders: {
+				supported: true
+			}
+		};
+	}
+	return result;
+});
+
+connection.onInitialized(() => {
+	if (hasConfigurationCapability) {
+		// Register for all configuration changes.
+		connection.client.register(DidChangeConfigurationNotification.type, undefined);
+	}
+
+	if(hasDidChangeWatchedFilesCapability) {	
+		const option : DidChangeWatchedFilesRegistrationOptions = {watchers: [{globPattern: '**/*.lg'}]};
+		connection.client.register(DidChangeWatchedFilesNotification.type, option);
+	}
+
+	if (hasWorkspaceFolderCapability) {
+		connection.workspace.onDidChangeWorkspaceFolders(_event => {
+			workspaceFolders = workspaceFolders?.filter(workspaceFolder => !_event.removed.includes(workspaceFolder));
+			_event.added.forEach(folderAdded => workspaceFolders?.push(folderAdded));
+			util.triggerLGFileFinder(workspaceFolders!);
+			connection.console.log('Workspace folder change event received.');
+		});
+	}
+});
+
+// The example settings
+interface ExampleSettings {
```

Comment:
change the naming of ExampleSettings. You can use LGSettings

Reply:
fixed